src/client/app/services/github.service.ts

```diff
@@ -119,10 +119,21 @@ export class GitHubService {
         let url = `${this._baseUrl}/gists`;
         if (!(id == null)) {
             url += `/${id}`;
-            return this._request.patch<IGist>(url, body, ResponseTypes.JSON, this._headers);
+            return this.gist(id)
+                .mergeMap(gist => {
+                    let oldFilename = Object.keys(gist.files)[0];
+                    let newFilename = Object.keys(files)[0];
+                    body.files = {};
```

Comment:
Feels weird to reset it here.  Should this not have been set on line 116 above, then?  And have body.files get set in the "else" clause?...